guides/source/security.md

```diff
@@ -1118,34 +1115,49 @@ class PostsController < ApplicationController
     policy.base_uri "https://www.example.com"
   end
 end
+```
+
+Or it can be disabled:
 
+```ruby
+# Disabling the global CSP
+class LegacyPagesController < ApplicationController
+  content_security_policy false, only: :index
+end
+```
+
+Use lambdas to inject per-request values, such as account subdomains in a
+multi-tenant application:
+
+```ruby
 # Using mixed static and dynamic values
 class PostsController < ApplicationController
   content_security_policy do |policy|
     policy.base_uri :self, -> { "https://#{current_user.domain}.example.com" }
   end
 end
-
-# Disabling the global CSP
-class LegacyPagesController < ApplicationController
-  content_security_policy false, only: :index
-end
 ```
 
 #### Reporting Violations
 
-Use the `content_security_policy_report_only`
-configuration attribute to set
+Enable the
+[report-uri](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/report-uri)
+directive to report violation to the specified URI:
+
+```ruby
+Rails.application.config.content_security_policy do |policy|
+  policy.report_uri "/csp-violation-report-endpoint"
+end
+```
+
+When migrating legacy content set the
 [Content-Security-Policy-Report-Only](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy-Report-Only)
-in order to report only content violations for migrating
-legacy content
+response header to only report content violations:
 
 ```ruby
 # config/initializers/content_security_policy.rb
 Rails.application.config.content_security_policy_report_only = true
-```
 
-```ruby
```

Comment:
When code snippets are from different contexts (e.g. `config/initializers/content_security_policy.rb` and `app/controllers/posts_controller.rb`), I prefer to keep the code blocks separate to provide visual distinction in the rendered output.  Also, it gives each snippet a dedicated "Copy" button.

(I did some work toward that in #40922.)

Reply:
Yes, good point. I agree.